treatment-service/services/experiment_service.go

```diff
@@ -128,38 +128,37 @@ func (es *experimentService) filterByLookupOrder(
 	filtered := matches
```

Comment:
I'm wondering if the above `// Stop search when we have at least 1 match` is still relevant here. Feels like it should be shifted down 🤔

Reply:
I missed this comment - not sure it had much significance previously either. Removed it. Thanks for pointing out.